### PR TITLE
ignore escaped quote-characters resolves #74

### LINKED ISSUE #74: In `TextObjectQuotedString` escaped characters should be ignored
For example I have a string with quotes, which contains escaped quotes:

and I press `ci'` inside of this string then it should change the whole string inside of the quotes and ignore the escaped quotes.

```
lorem 'ipsum \' asdf\' laskdfj' more text
         ^
```
expected
```
lorem '' more text
```
Currently it only replaces the string next to the escaped quote.

It should work the same for strings in `""` and backticks.

### PR DESCRIPTION
**Pull request by DanEEStar**
This commit should fix #74

### CODE REVIEW

File: src/TextObjects/QuotedString.ts [hunk 1]
```diff
@@ -38,6 +38,10 @@ export class TextObjectQuotedString extends TextObject {
 
         while (characterIndex >= 0) {
             if (lineText[characterIndex] === this.quoteCharacter) {
+                if(lineText[characterIndex-1] === '\\') {
```
Comment (aioutecism) [inline src/TextObjects/QuotedString.ts]: How about define a static property at the very beginning of this class?

```ts
export class TextObjectQuotedString extends TextObject {

    private static escapeCharacter: string = '\\';

    private quoteCharacter: string;
    private adjustedAnchor: Position;

    // ...
}
```

File: src/TextObjects/QuotedString.ts [hunk 2]
```diff
@@ -52,6 +56,10 @@ export class TextObjectQuotedString extends TextObject {
 
         while (characterIndex < lineText.length) {
             if (lineText[characterIndex] === this.quoteCharacter) {
+                if(lineText[characterIndex-1] === '\\') {
+                    characterIndex++;
+                    continue;
+                }
```
Comment (aioutecism) [inline src/TextObjects/QuotedString.ts]: We only appended a condition to previous `if` statement. So how about combine them together?

```ts
if (lineText[characterIndex] === this.quoteCharacter &&
    lineText[characterIndex - 1] !== TextObjectQuotedString.escapeCharacter) {
    // ...
}
```

**Comment by aioutecism** [comment]
Thanks for the fix!
I missed that.

**Comment by DanEEStar** [comment]
@aioutecism adopted changes from the comments

**Comment by aioutecism** [comment]
New versoin

**Pull request merged**